src/Application/Digital/Commands/ShareEntry.cs

```diff
@@ -0,0 +1,173 @@
+            
+            await GrantPermission(entry, user, allowOperations, request.ExpiryDate, cancellationToken);
+
+            if (entry.IsDirectory)
+            {
+                // Update permissions for child Entries
+                var path = entry.Path.Equals("/") ? entry.Path + entry.Name : $"{entry.Path}/{entry.Name}";
+                var pattern = $"{path}/%";
+                var childEntries = _context.Entries
+                    .Include(x => x.Owner)
+                    .Where(x => x.Id != entry.Id 
+                                && (x.Path.Equals(path) || EF.Functions.Like(x.Path, pattern))
+                                && x.OwnerId == entry.OwnerId)
+                    .ToList();
+                foreach (var childEntry in childEntries)
+                {
+                    var childAllowOperations = GenerateAllowOperations(request, childEntry.IsDirectory);
+                    await GrantPermission(childEntry, user, childAllowOperations, request.ExpiryDate, cancellationToken);
+                }
+            }
+            await _context.SaveChangesAsync(cancellationToken);
+            
+            var result = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == request.EntryId
+                    && x.EmployeeId == request.UserId, cancellationToken);
+            return _mapper.Map<EntryPermissionDto>(result);
+        }
+
+        private async Task GrantPermission(
+            Entry entry,
+            User user,
+            string allowOperations,
+            DateTime expiryDate,
+            CancellationToken cancellationToken)
+        {
+            var existedPermission = await _context.EntryPermissions.FirstOrDefaultAsync(x =>
+                    x.EntryId == entry.Id
+                    && x.EmployeeId == user.Id, cancellationToken);
+            
+            if (existedPermission is null)
+            {
+                var entryPermission = new EntryPermission
+                {
+                    EmployeeId = user.Id,
+                    EntryId = entry.Id,
+                    AllowedOperations = allowOperations,
+                    ExpiryDateTime = LocalDateTime.FromDateTime(expiryDate),
+                    Employee = user,
+                    Entry = entry,
+                };
+                await _context.EntryPermissions.AddAsync(entryPermission, cancellationToken);
+            }
+            else
+            {
+                existedPermission.AllowedOperations = allowOperations;
+                existedPermission.ExpiryDateTime = LocalDateTime.FromDateTime(expiryDate);
+                _context.EntryPermissions.Update(existedPermission);
+            }
+        }
+
+        private static string GenerateAllowOperations(Command request, bool isDirectory)
+        {
+            if (request is { CanView: false, CanDownload: false, CanUpload: false, CanChangePermission: false })
```

Comment:
I don't think this is needed, either way this method's gonna return empty string right?

Reply:
No, I include it because it return null instead. And it's not a nullable prop in db. So 500 to the client.

Reply:
ok makes sense